lints/lint_dnsname_bad_character_in_label.go

```diff
@@ -0,0 +1,59 @@
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+	"regexp"
+	"strings"
+)
+
+type DNSNameProperCharacters struct {
+	// Internal data here
+	CompiledExpression *regexp.Regexp
+}
+
+func (l *DNSNameProperCharacters) Initialize() error {
+	const dnsLabelRegex = "^[A-Za-z0-9*_-]+$"
+	l.CompiledExpression = regexp.MustCompile(dnsLabelRegex)
+	return nil
+}
+
+func (l *DNSNameProperCharacters) CheckApplies(c *x509.Certificate) bool {
```

Comment:
This seems like it should only apply to certificates with domains. I don't think this applies to CA names, etc.